src/middlewared/middlewared/plugins/enclosure_/slot_mappings.py

```diff
@@ -142,290 +238,320 @@ def get_slot_info(model, enclosures=None):
                     'product': {
                         'eDrawer4048S1': {
                             # 1 - 24
-                            i: {'orig_slot': i, 'mapped_slot': i} for i in range(1, 25)
+                            i: {SYSFS_SLOT_KEY: i - 1, MAPPED_SLOT_KEY: i} for i in range(1, 25)
                         },
                         'eDrawer4048S2': {
                             # 25 - 48
-                            i: {'orig_slot': i, 'mapped_slot': j} for i, j in zip(range(1, 25), range(25, 49))
+                            i: {SYSFS_SLOT_KEY: i - 1, MAPPED_SLOT_KEY: j} for i, j in zip(range(1, 25), range(25, 49))
```

Comment:
These changed because sysfs reports the device slot numbers that the HBA reports by default. The device slot numbers for 99% of the HBAs/enclosures that we sell all start at index 0. The only exception is the ES102G2 which starts at 1. So the changes here are fixing a problem 😄 I've added more docs to try and explain this.